File: KickStart/November_13_2021/Q1.cpp
#include <iostream>
#include <string>

int minTransform(std::string s, std::string f) {
    int total = 0;
    for (auto character : s) {
        if (f.find(character) == std::string::npos) {
            int min = 27;
            for (auto character2 : f) {
                int distance = abs(character - character2);
                if (distance > 13) {
                    distance = 26 - distance;
                }
                if (distance < min) {
                    min = distance;
                }
            }
            total += min;
        }
    }
    return total;
}

int main() {
    int numTestCases;
    std::cin >> numTestCases;
    for (int i = 0; i < numTestCases; i++) {
        std::string s;
        std::string f;
        std::cin >> s >> f;

        std::cout << "Case #" << i + 1 << ": " << minTransform(s, f) << "\n";
    }
}

File: KickStart/November_13_2021/Q1.rs
use std::io;

fn minTransform (s: &str, f: &str) -> i32 {
    let mut total = 0;
    for c in s.chars() {
        if !f.find(c).is_some() {
            let mut min = 27;
            for c2 in f.chars() {
                let mut distance = (c as i32 - c2 as i32).abs();
                if distance > 13 {
                    distance = 26 - distance;
                }
                if distance < min {
                    min = distance;
                }
            }
            total += min;
        }
    }
    return total;
}

fn main() -> io::Result<()> {
    let mut input = String::new();
    io::stdin().read_line(&mut input)?;
    let test_cases: u32 = input.trim().parse().unwrap();

    for test_case in 1..test_cases + 1 {
        let mut result = 0;
        let mut s = String::new();
        let mut f = String::new();
        io::stdin().read_line(&mut s)?;
        io::stdin().read_line(&mut f)?;
        // remove delimiter
        s.pop();
        f.pop();
        result = minTransform(&s, &f);
        println!("Case #{}: {}", test_case, result);
    }
    return Ok(());
}